### PR TITLE
Restrict command access based on app name roles

### PR DESCRIPTION
**Pull request by chaselee**
Potential solution for https://github.com/daemonsy/hubot-heroku/issues/9

### CODE REVIEW

**Comment by jivinivan** [comment]
This is definitely useful!

**Comment by chaselee** [comment]
@daemonsy, this should be all set. Let me know if you have questions or thoughts!

**Comment by daemonsy** [comment]
Thanks! :heart_eyes: I'll look into this over the weekend

**Comment by chaselee** [comment]
Great! Let me know if you have questions. I tried to keep things simple. My team's been using it since yesterday, and so far so good. It's only turned on by setting a config var, so it shouldn't affect current users. :smiley:

**Comment by chaselee** [comment]
Hey @daemonsy! Let me know if you have questions on these pulls. Happy to help.

**Comment by daemonsy** [comment]
Hey @chaselee thanks for this! Sorry it took so long to get back to you. This man just got married a while ago, trying to get used to the new life. Ok and this is last time I'll use this excuse =)


Looks good to me. Using the app name as the role makes sense. Have you looked into test cases for this change? The hubot test helper on this library is pretty "mockish" though, let me know if it gets in the way, happy to help.

Thanks for using the extension, happy chat ops :smile:

File: README.md [hunk 1]
```diff
@@ -31,6 +41,8 @@ hubot heroku config:set my-hubot HUBOT_ADMIN=dr_evil
 # Muhaha, now I'm to use hubot's other commands to take over the world
 ```
 
+You can also avoid this if you are using auth as described above, in which case you can ensure only admins have the role necessary to set config variables on the hubot instance.
```
Comment (daemonsy) [inline README.md]: Mental note to self: Also add a warning about disabling hubot-brain dumping before release

**Comment by chaselee** [comment]
added tests in #16. will do this one next.

**Comment by chaselee** [comment]
Added tests @daemonsy. Let me know what you think.

**Comment by chaselee** [comment]
any thoughts @daemonsy?

**Comment by chaselee** [comment]
FYI @daemonsy this has been updated with the new dyno and restart commands

**Comment by chaselee** [comment]
@daemonsy have you had a chance to look at this? Anything I can answer?

**Comment by daemonsy** [comment]
Hey sorry for the delay, thanks for being so patient with the PR. It's looking good. :+1:

Merging and releasing it after putting it to test on our own hubot :smile:

For some context, in our team we have developers who have the "devops" badge and  non devops members. The badged guys are in charge of the potentially dangerous stuff, such as upgrading DBs, running generic heroku commands etc. 

The initial design of the bot is had a flat permissions system because its assumption is that everyone can run these commands and the commands should be safe. Well you still can run a migration that deletes everything, but that should be handled at the code review level.

I'm mulling over putting some work into a tiered permission system or just guard individual commands with roles to keep the objectives above. Either way, your PR still makes sense because right now the commands can be accessed by anyone in the chat room, including guests.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,16 @@ Deployment usually involves some form of CI process. Hence it is best suited for
 
 This robot is focused on letting you run auxiliary commands around the heroku system, so developers don't have to be given production access to independently manage deployments.
 
+## Auth
+You can restrict command usage to specific roles using the [hubot-auth](https://github.com/hubot-scripts/hubot-auth) package. Role names take the form `heroku-<app>`.
+
+To enable auth:
+
+1. `npm install hubot-auth --save`
+2. Add `hubot-auth` to `external-scripts.json` (e.g. `["hubot-auth", "some-other-plugin"]`)
```
Comment (daemonsy) [inline README.md]: Mental note: What's the best way to specify this dependency? If users forget to add them, hubot actually deploys but fails on starting (at least on Heroku)

**Pull request merged**